bazel/foreign_cc/luajit.patch

```diff
@@ -65,8 +65,8 @@ index 0000000..9c71271
 +
 +    os.environ["MACOSX_DEPLOYMENT_TARGET"] = "10.6"
 +    os.environ["DEFAULT_CC"] = os.environ.get("CC", "")
-+    os.environ["TARGET_CFLAGS"] = os.environ.get("CFLAGS", "")
-+    os.environ["TARGET_LDFLAGS"] = os.environ.get("CFLAGS", "")
++    os.environ["TARGET_CFLAGS"] = os.environ.get("CFLAGS", "") + " -fno-function-sections -fno-data-sections"
++    os.environ["TARGET_LDFLAGS"] = os.environ.get("CFLAGS", "") + " -fno-function-sections -fno-data-sections"
```

Comment:
Isn't there an old cut/paste error carried over?
Why would os.environ["TARGET_LDFLAGS"] not be os.environ.get("LDFLAGS" ...?

Also, isn't LDFLAGS the linker flag and then no need to have these -f that are compiler flags. (aren't linker flags prefixed with -l and -Wl or something?)

Reply:
I assumed this was intentional, but perhaps ask @lizan or on the original PR (https://github.com/envoyproxy/envoy/pull/6168)?

Reply:
Yeah, looking at [LuaJIT's Makefile](https://github.com/LuaJIT/LuaJIT/blob/v2.1.0-beta3/src/Makefile#L211), it's invoking compiler in place of linker (`TARGET_LD= $(CROSS)$(CC)`), so I believe that using `CFLAGS` is correct here.

Reply:
But isn't the override incorrect? LDFLAGS is supposed to override TARGET_LDFLAGS.
c.f. comment from the Makefile
https://github.com/LuaJIT/LuaJIT/blob/v2.1.0-beta3/src/Makefile#L181